pkg/bot.go

```diff
@@ -395,6 +426,22 @@ func (b *bot) deletePreviewSite(args ResponseRequest, verboseErrors bool) string
 	return strings.Join(msgs, "\n\n")
 }
 
+func (b *bot) hasPreviewSiteCapability(email, org string) bool {
+	ctx := context.Background()
+	metaCtx := grpcmeta.AppendToOutgoingContext(ctx, "x-ddev-workspace", org)
+	caps, err := b.authClient.ListCapabilities(metaCtx, &pbAdmin.ListCapabilitiesRequest{Email: email})
+	if err != nil {
+		klog.Errorf("failed querying capabilities for user %v: %v", email, err)
+		return false
+	}
+	for _, c := range caps.Capabilities {
+		if c == pbAdmin.Capability_SiteEditor {
```

Comment:
I think if someone has capability `SiteEditor`, they could be trusted to create preview sites, @drekle do you have any alternative recommendations?